mmv1/products/iam2/api.yaml

```diff
@@ -0,0 +1,152 @@
+# Copyright 2022 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: IAM2
+display_name: Cloud IAM
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://iam.googleapis.com/v2beta/
+scopes:
+  - https://www.googleapis.com/auth/iam
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Identity and Access Management (IAM) API
+    url: https://console.cloud.google.com/apis/library/iam.googleapis.com/
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: True
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'DenyPolicy'
+    min_version: beta
+    base_url: policies/{{parent}}/denypolicies
+    create_url: policies/{{parent}}/denypolicies?policyId={{name}}
+    description: |
+      Represents a collection of denial policies to apply to a given resource.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Permissions supported in deny policies':
+          'https://cloud.google.com/iam/docs/deny-permissions-support'
+      api: 'https://cloud.google.com/iam/docs/reference/rest/v2beta/policies'
+    properties:
+      - !ruby/object:Api::Type::String
+         name: 'name'
+         description: |
+           The attachment point is identified by its URL-encoded full resource name.
+         required: true
+         input: true
+         url_param_only: true
+      - !ruby/object:Api::Type::String
+         name: 'parent'
+         description: |
+           The attachment point is identified by its URL-encoded full resource name.
+         required: true
+         input: true
+         url_param_only: true
+      - !ruby/object:Api::Type::String
+         name: 'displayName'
+         description: |
+           The display name of the rule.
+      - !ruby/object:Api::Type::String
```

Comment:
We _may_ be able to use the `::Fingerprint` type if making this an output is undesirable. This is to make sure we include it in the update request, right?

It's not the biggest deal to include- we've considered it safe to blank out an etag in a minor version in the past.